src/components/dialog/Dialog.spec.js

```diff
@@ -69,6 +69,19 @@ describe('BDialog', () => {
         expect(wrapper.vm.onConfirm).toHaveBeenCalled()
     })
 
+    it('close on async confirm', async () => {
+        wrapper.setProps({
+            onConfirm: jest.fn((confirm, {StartLoading}) => {
+                StartLoading()
```

Comment:
@zhanghengxin @wesdevpro
I am thinking enabling the "Loading" behavior in `Dialog` by default should not harm. I mean clients does not have to call `StartLoading` themselves, but `Dialog` sets itself the "Loading" state whenever `confirm` is called. The "Loading" state will be cancelled when `close` or `cancelLoading` (or `cancelConfirm`) is called.

Reply:
It makes sense, it can completely be an internal state.

Reply:
> @zhanghengxin @wesdevpro I think enabling the "Loading" behavior in `Dialog` by default should not harm. I mean clients do not have to call `StartLoading` themselves, but `Dialog` sets itself the "Loading" state whenever `confirm` is called. The "Loading" state will be cancelled when `close` or `cancelLoading` (or `cancelConfirm`) is called.

I realized I was wrong about the above comment. It may be a breaking change for some applications that intentionally keep the `Dialog` opened after the confirm button is clicked, which will end up with being stuck in the loading state unless they change their code.

@zhanghengxin Could you revert to your initial idea of calling `startLoading` explicitly?

Reply:
@kikuomax 

I have restored it.